### PR TITLE
Add automatic image detection

### LINKED ISSUE #43: Automatic image detection
We can look at a binary and if it looks like an image, we can automatically show it as said image.

### PR DESCRIPTION
**Pull request by jannikbecher**
This is my first time contributing to this project, and I'm excited to make my first pull request. Please let me know if there's anything I can improve!

Closes #43.

Detects image type by its magic number and renders the image as a tabbed output.
kino_pr_add_automatic_image_detection

### CODE REVIEW

File: lib/kino/utils.ex [hunk 1]
```diff
@@ -40,4 +40,21 @@ defmodule Kino.Utils do
          do: true,
          else: (_ -> false)
   end
+
+  @doc """
+  Gets the image type of a file by the magic number
+  """
+  @spec get_image_type(bitstring()) :: {:ok, Kino.Image.common_image_type()} | :not_found
```
Comment (jonatanklosko) [inline lib/kino/utils.ex:47]: I'd do this :)

```suggestion
  @spec image_type(binary()) :: Kino.Image.common_image_type() | nil
```

**Comment by jonatanklosko** [comment]
@jannikbecher fantastic, thanks for the PR!

**Comment by jannikbecher** [comment]
@jonatanklosko thanks for this great framework! It was a pure joy try to figure out what needs to be changed :)

**Comment by jannikbecher** [comment]
sorry for the commit mess..
still learning the pull request workflow

**Comment by jonatanklosko** [comment]
No worries, take your time. We will squash it all into a single commit when merged anyway, so it's fine  :)

**Comment by jannikbecher** [comment]
Should be good now, thanks for the patience :)
One question: do you use the web interface for code reviews?

**Review by jonatanklosko — APPROVED** [review]
Perfect!

> One question: do you use the web interface for code reviews?

Yes :)

**Pull request merged**